src/server/express/a2a_express_app.ts

```diff
@@ -32,6 +32,20 @@ export class A2AExpressApp {
         const router = express.Router();
         router.use(express.json(), ...(middlewares ?? []));
 
+        router.use((err, req, res, next) => {
+            // Check if it's the specific JSON parsing error
+            if (err instanceof SyntaxError && 'body' in err) {
+                const a2aError = A2AError.parseError('Failed to parse JSON request.');
+                const errorResponse: JSONRPCErrorResponse = {
+                                jsonrpc: '2.0',
+                                id: req.body?.id || null, // Use original request ID if available
```

Comment:
It seems like the spec says it must be nil if request wasn't parsed: https://a2a-protocol.org/latest/specification/#6112-jsonrpcresponse-object.